pkg/controller/statusmanager/status_manager.go

```diff
@@ -49,8 +53,49 @@ type StatusManager struct {
 	relatedObjects []configv1.ObjectReference
 }
 
-func New(client client.Client, name, version string) *StatusManager {
-	return &StatusManager{client: client, name: name, version: version}
+func New(client client.Client, mapper meta.RESTMapper, name, version string) *StatusManager {
+	return &StatusManager{client: client, mapper: mapper, name: name, version: version}
+}
+
+// deleteRelatedObjects checks for related objects attached to ClusterOperator and deletes
+// whatever is not been rendered from manifests. This is a mechanism to cleanup objects
+// that are no longer needed and are probably present from a previous version
+func (status *StatusManager) deleteRelatedObjectsNotRendered(co *configv1.ClusterOperator) {
+	if status.relatedObjects != nil {
+		for _, current_obj := range co.Status.RelatedObjects {
+			var found bool = false
+			for _, rendered_obj := range status.relatedObjects {
+				found = reflect.DeepEqual(current_obj, rendered_obj)
+
+				if found {
+					break
+				}
+			}
+			if !found {
+				gvr := schema.GroupVersionResource{
+					Group:    current_obj.Group,
+					Resource: current_obj.Resource,
+				}
+				gvk, err := status.mapper.KindFor(gvr)
+				if err != nil {
+					log.Printf("Error getting GVK of object for deletion: %v", err)
+					co.Status.RelatedObjects = append(co.Status.RelatedObjects, current_obj)
```

Comment:
`co.Status.RelatedObjects` gets overwritten as soon as you return from this function; you need to append it to `status.relatedObjects`.